api/v01/api_base.rb

```diff
@@ -52,7 +52,7 @@ def self.count_route_locations(params)
       def self.count_matrix_locations(params)
         src_size = count_locations(params[:src])
         dst_size = params[:dst] ? count_locations(params[:dst]) : src_size
-        src_size * dst_size
+        [src_size, dst_size].max
```

Comment:
For clustering purpose we may need rectangular matrices for exemple 2 (depots) * 5000 (items) before effectively. I was satisfied yesterday by the Math.sqrt(src, dst) could we return to this solution ?

Reply:
I think even for clustering, the maximum value is what we need to control/limit

Reply:
Ok, after discussion, it was a mixture of two concepts: one is a count of matrix elements used for user queries and the other is a limitation on the size of a matrix. This is specific to the matrix because of its dimensional aspect. 

We now have two methods.